scripts/pre-install.js

```diff
@@ -87,26 +85,30 @@ const REQUIRED_VERSION = {
     revision: 1,
 };
 
-function downloadFile(url, destinationFile) {
-    console.log(`Downloading nrfjprog from ${url} to ${destinationFile}...`);
+function downloadFile(fileid, destinationFile) {
+    console.log(`Downloading nrfjprog from ${DOWNLOAD_URL} with fileid ${fileid} to ${destinationFile}...`);
 
     const destinationDir = path.dirname(destinationFile);
     return sander.mkdir(destinationDir)
-        .then(() => new Promise((resolve, reject) => {
+        .then(() => new Promise(async (resolve, reject) => {
```

Comment:
I would suggest to rewrite this to a chain of promises by first executing the axios request and chain the file creation afterwards.